spec/compiler/parser/parser_spec.cr

```diff
@@ -163,6 +163,20 @@ describe "Parser" do
   assert_syntax_error "def foo!=; end", "unexpected token: !="
   assert_syntax_error "def foo?=(x); end", "unexpected token: ?"
 
+  # #5895
+  %w(
+    begin nil true false yield with abstract
```

Comment:
The manual list is bad enough, let's not put in two places. Just add a few examples.

Reply:
And don't use any interpolation in the examples because tests should be "unrolled"

Reply:
Why bad enough? `parser_spec.cr` uses such a list in 10 places at least. And I found the implementation bug by this test before commiting in fact.